infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/workflows/HealNodeWorkflow.java

```diff
@@ -33,20 +32,15 @@
     public HealNodeWorkflow(HealNodeRequest healNodeRequest) {
         super(new AddNodeRequest(healNodeRequest.getEndpoint()));
         this.request = healNodeRequest;
+        this.actions = ImmutableList.of(new HealNodeToLayout(),
+                new RestoreRedundancyAndMergeSegments());
     }
 
     @Override
     public String getName() {
         return HEAL_NODE.toString();
     }
 
-    @Override
-    public List<Action> getActions() {
```

Comment:
Did you see any significant time/resource on multiple list creation or it was just proactive improvement?

Reply:
It was a required item in #1350 
I should have done it in the same PR but overestimated the number of changes required.

Reply:
Thanks. I was just curious.

Reply:
This is not a performance issue. Its incorrect behavior. Calling getActions twice would return a new set of actions, i.e. the state of the previous actions is lost.